src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -1852,6 +1852,7 @@ void Session::send_bind_message()
         if (auto migrated_partition = get_migration_store()->get_migrated_partition()) {
             bind_json_data["migratedPartition"] = *migrated_partition;
         }
+        bind_json_data["sessionReason"] = static_cast<uint64_t>(get_session_reason());
```

Comment:
@kmorkos - should this be a string value, since most of our protocol "enum" values are strings instead of ints?

Reply:
i have no strong preference. thoughts @yaseenmongo? 

regardless, it would be nice to represent a "normal" session as the zero-value of whatever type we choose for backwards compatibility

Reply:
A normal session has a value of zero indeed

Reply:
I don't have a preference either. Since this is a wire protocol detail, I would generally say an int is better just for the slight efficiency gain, but it's not that much data to really matter. I'll defer to you folks 🙂

Reply:
it is an integer then 🙂

Reply:
One note, one potential benefit of using an `int` is that we could use convert it as a bitmask in the future **if** we ever needed to communicate multiple reasons why a client is reconnecting to sync e.g. `ClientReset` and `SomeOtherReason`. But I'm not sure if that will ever become a requirement.

Reply:
I'm OK with that